### DISCUSSION TITLE
revert back to 1.6.2

### DISCUSSION THREAD

**Discussion by Scherzingetorix** [tiny-pilot/tinypilot#1731] [General]
Hi,

Is there a relatively simple way to revert to version 1.6.2 on my self-built Raspberry Pi 4 solution? I've been using this version for a long time without any issues. Recently, I updated, and it became slower over time. Since the last update, it's unusable because MJPEG compression doesn't work, and the H.264 freezes after a few minutes. So, I'd like to roll back to version 1.6.2. How can I do this? Could you provide a step-by-step guide? 

Thanks in advance!

**Comment by mtlynch** [comment]
We don't have great support for jumping back to old versions, as we'd rather understand and fix what's wrong rather than have people roll back to versions from years ago.

This *might* work, but I think it's likely to hit issues:

```bash
curl \
  --silent \
  --show-error \
  https://raw.githubusercontent.com/tiny-pilot/tinypilot/1.6.2/quick-install | \
    bash - && \
  sudo reboot
```

Alternatively, this might work:

```bash
# I know you want 1.6.2, but this is the earliest version that
# supports this install method.
TARGET_VERSION="1.8.1"
cd $(mktemp -d) && \
  git clone https://github.com/tiny-pilot/tinypilot.git . && \
  git checkout "${TARGET_VERSION}" && \
  ./dev-scripts/install-from-source
```

Again, these aren't supported, so I can't offer further assistance if they don't work, but I wanted to share some pointers of things that might work.

We're also happy to help you diagnose what could be going wrong in the current version.

**Reply by Scherzingetorix** [reply]
Thank you for the quick response and the effort you put into it. Unfortunately, neither solution worked as it encountered errors with various things, as you mentioned, which was likely. After that, I reflashed the Raspberry Pi with a Bullseye distribution and did a clean install of TinyPilot. The situation is almost the same: in MJPEG mode, the image disappears after a few minutes, and in H.264 mode, the lower quarter of the image occasionally flickers. By flickering, I mean that the image disappears in a strip or changes color briefly. In the log, I found only this:      
132  Jan 30 13:01:51 raspberrypi launch[532]: -- ERROR [10.057    stream] -- Could not obtain the requested format=JPEG; driver gave us MJPEG
136  Jan 30 13:01:51 raspberrypi launch[532]: -- ERROR [10.059    stream] -- Device doesn't support setting of HW encoding quality parameters
144  Jan 30 13:01:51 raspberrypi launch[532]: -- ERROR [10.367    stream] -- Can't decompress JPEG: Unsupported marker type 0x15. 

Do you have any ideas on what to do with this? At first glance, it seems like some kind of driver error, which would explain the flickering image in H.264 mode. However, I really have no idea what to do with this.

Thank you in advance!

**Reply by mtlynch** [reply]
Sorry to hear it didn't work?

Can you [file a bug](https://github.com/tiny-pilot/tinypilot/issues/new?assignees=&labels=bug&projects=&template=bug_report.md&title=) and include the details of the hardware you're using (which Pi version, which HDMI capture)?
